frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/scheduler/KafkaService.java

```diff
@@ -29,13 +30,16 @@ public KafkaService(File pathToYamlSpecification) throws Exception {
                 DefaultScheduler.newBuilder(YAMLServiceSpecFactory.generateServiceSpec(rawServiceSpec));
         schedulerBuilder.setPlansFrom(rawServiceSpec);
 
-        /* Upgrade */
-        new KafkaConfigUpgrade(schedulerBuilder.getServiceSpec());
-        CuratorStateStoreFilter stateStore = new CuratorStateStoreFilter(schedulerBuilder.getServiceSpec().getName(),
+        /* Post Upgrade for version 1.1.21 */
+        CuratorStateStore stateStore = new CuratorStateStore(schedulerBuilder.getServiceSpec().getName(),
                 DcosConstants.MESOS_MASTER_ZK_CONNECTION_STRING);
-        stateStore.setIgnoreFilter(RegexMatcher.create("broker-[0-9]*"));
-        schedulerBuilder.setStateStore(stateStore);
-        /* Upgrade */
+        StringMatcher oldTaskNameFilter = RegexMatcher.create("broker-[0-9]*");
```

Comment:
Current master has code to autodetect if Configuration upgrade needed or not. If upgrade needed, it does and while doing that keeps taskInfo backups. If new Kafka fails, you can revert back to old Kafka.

It is optional, whether cleanup backups or not after upgrade.  Default is keep backups. Backups are not useful after new Kafka is running correctly. So, we delete them. 

PS: if we do not delete those backup taskInfos, should use CuratorFilter, otherwise those will confuse recovery process, and placement rule calculations.